packages/snap/src/index.ts

```diff
@@ -55,75 +53,93 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) =>
+              text(
+                `The suggested gas currency for your tx is ${suggestedFeeCurrency}`,
+              ),
+              text(
+                `If you would like to use a different gas currency, please enter it below`,
+              ),
+              text(`Otherwise, press submit`),
             ]),
-            placeholder: `'cusd', 'ceur', 'creal', 'celo'`
-          }
-        })
+            placeholder: `'cusd', 'ceur', 'creal', 'celo'`,
+          },
+        });
 
-        if ( // TODO find a cleaner way to do this, probably use an enum 
+        if (
+          // TODO find a cleaner way to do this, probably use an enum
           overrideFeeCurrency === 'cusd' ||
           overrideFeeCurrency === 'ceur' ||
           overrideFeeCurrency === 'creal' ||
-          overrideFeeCurrency === 'celo'  
+          overrideFeeCurrency === 'celo'
         ) {
-          tx.feeCurrency = getFeeCurrencyAddressFromName(overrideFeeCurrency, network.name)
+          tx.feeCurrency = getFeeCurrencyAddressFromName(
+            overrideFeeCurrency,
+            network.name,
+          );
         }
 
         try {
-          const txReceipt = await sendTransaction(tx, wallet)
+          const txReceipt = await sendTransaction(tx, wallet);
           await snap.request({
             method: 'snap_dialog',
             params: {
               type: 'alert',
               content: panel([
                 text(`Your transaction succeeded!`),
-                copyable(`${network.explorer}/tx/${txReceipt.transactionHash}`)
-              ])
-            }
-          })
+                copyable(`${network.explorer}/tx/${txReceipt.transactionHash}`),
+              ]),
+            },
+          });
         } catch (error) {
+          let message = JSON.stringify(error);
+
+          if (isInsufficientFundsError(error)) {
+            message =
+              'Oops! Looks like the chosen gas currency is not sufficient to complete the operation. Please try again using another currency.';
+          }
+
           await snap.request({
             method: 'snap_dialog',
             params: {
               type: 'alert',
               content: panel([
                 text(`Your transaction failed!`),
-                text(`error: ${JSON.stringify(error)}`)
-              ])
-            }
-          })
+                text(`error: ${message}`),
+              ]),
+            },
+          });
         }
       }
 
     default:
-      throw new Error('Method not found.')
+      throw new Error('Method not found.');
   }
 }
 
 async function getNetworkConfig(): Promise<Network> {
   const chainId = await ethereum.request({ method: 'eth_chainId' }) as string
-  return getNetwork(chainId) // TODO
+  return getNetwork(chainId)
 }
 
 async function sendTransaction(tx: CeloTransactionRequest, wallet: CeloWallet) {
-  
+  tx.value = BigNumber.from(tx.value) // todo investigate why we get a hex error w/o this.
```

Comment:
It seems that since `CeloTransactionRequest` from celo-tools extends `providers.TransactionRequest` which returns a HEX by default.

celo-tools has `handleNumber()` that might be helpful to parse the hex into a number

Reply:
thanks for this feedback, ok since handleNumber is not exported, copying it over locally